source/common/quic/envoy_quic_client_stream.h

```diff
@@ -51,6 +51,8 @@ class EnvoyQuicClientStream : public quic::QuicSpdyClientStream,
   void OnConnectionClosed(quic::QuicErrorCode error, quic::ConnectionCloseSource source) override;
 
   void clearWatermarkBuffer();
+  // Enables the QUIC stream to decode HTTP/3 Datagrams and makes it encode DATA frames into HTTP/3
+  // Datagrams or Capsules.
```

Comment:
Can we expand on what datagram support entails? In particular, it looks like when this method is called, http_datagram_handler_ will be initialized. Once this happens, any calls to encodeData are expected to contain capsules which will be sent along as HTTP data grams. (I think that's right)

As such, I'm not sure that this method enables datagram *support* (which sounds like it could be used or not, depending on context). Instead I think this enables the capsule protocol handling. So perhaps something like `useCapsuleProtocol()` or some such?